rs-react-app/src/components/DetailPanel.tsx

```diff
@@ -0,0 +1,90 @@
+import { useEffect, useState } from 'react';
+import { useSearchParams } from 'react-router';
+import { fetchCharacterById } from '../api';
+import { Character } from '../types';
+import { Loader } from './Loader';
+
+export const DetailPanel = () => {
+  const [character, setCharacter] = useState<Character | null>(null);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState<string | null>(null);
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const id = searchParams.get('details');
+
+  useEffect(() => {
+    if (!id) return;
+
+    const fetch = async () => {
+      setLoading(true);
+      setError(null);
+      try {
+        const data = await fetchCharacterById(Number(id));
+        setCharacter(data);
+      } catch (err) {
+        if (err instanceof Error) setError(err.message);
+        else setError('Failed to load character.');
+      } finally {
+        setLoading(false);
+      }
+    };
+
+    fetch();
+  }, [id]);
+
+  const handleClose = () => {
+    searchParams.delete('details');
+    setSearchParams(searchParams);
+  };
+
+  if (loading) return <Loader />;
+  if (error) return <div className="text-rose-500">{error}</div>;
+  if (!character) return null;
+
+  return (
+    <div className="relative p-4 rounded-lg bg-slate-800">
```

Comment:
Just a small observation, when you scroll the page, detail item stays on the top, you can make it sticky to be always visible
```suggestion
    <div className="sticky top-20 p-4 rounded-lg bg-slate-800">
```